src/service-override/storage.ts

```diff
@@ -0,0 +1,146 @@
+import { Event } from 'vs/base/common/event'
+import { IDisposable } from 'vs/base/common/lifecycle'
+import { IStorage, IStorageDatabase, IStorageItemsChangeEvent, IUpdateRequest, Storage } from 'vs/base/parts/storage/common/storage'
+import { IEditorOverrideServices } from 'vs/editor/standalone/browser/standaloneServices'
+import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors'
+import { AbstractStorageService, IStorageService, StorageScope as VSStorageScope } from 'vs/platform/storage/common/storage'
+import { IUserDataProfile } from 'vs/platform/userDataProfile/common/userDataProfile'
+import { IAnyWorkspaceIdentifier } from 'vs/platform/workspace/common/workspace'
+
+export enum StorageScope {
+  APPLICATION = VSStorageScope.APPLICATION,
+  PROFILE = VSStorageScope.PROFILE,
+  WORKSPACE = VSStorageScope.WORKSPACE
+}
+
+export function StorageScopeToString (scope: StorageScope): string {
+  switch (scope) {
+    case StorageScope.APPLICATION: return 'APPLICATION'
+    case StorageScope.PROFILE: return 'PROFILE'
+    case StorageScope.WORKSPACE: return 'WORKSPACE'
+    default: return `UNKNOWN.${scope}`
+  }
+}
```

Comment:
is it used somewhere?

Reply:
That's a helper for provider implementer. Like in my case I don't really care about the actual scope and just need some string identifier to store it.

Reply:
I think I prefer to keep that user implementation specific code in the user code and not in that library